src/main/java/edu/cornell/kfs/kim/inquiry/CuPersonInquirableImpl.java

```diff
@@ -0,0 +1,136 @@
+import org.kuali.kfs.kns.web.ui.Field;
+import org.kuali.kfs.kns.web.ui.Section;
+import org.kuali.kfs.krad.bo.BusinessObject;
+import org.kuali.kfs.krad.util.GlobalVariables;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+
+import edu.cornell.kfs.kim.CuKimConstants.PersonInquirySections;
+import edu.cornell.kfs.kim.CuKimPropertyConstants;
+import edu.cornell.kfs.sys.CUKFSPropertyConstants;
+
+@SuppressWarnings("deprecation")
+public class CuPersonInquirableImpl extends PersonInquirableImpl {
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private static final Set<String> INQUIRY_SECTIONS_WITH_SUPPRESSED_FIELDS = Set.of(
+            PersonInquirySections.NAMES,
+            PersonInquirySections.ADDRESSES,
+            PersonInquirySections.PHONE_NUMBERS,
+            PersonInquirySections.EMAIL_ADDRESSES
+    );
+
+    private static final Set<String> UNMASKABLE_INQUIRY_FIELDS = Set.of(
+            CuKimPropertyConstants.EntityName.NAME_PREFIX,
+            CuKimPropertyConstants.EntityName.FIRST_NAME,
+            CuKimPropertyConstants.EntityName.MIDDLE_NAME,
+            CuKimPropertyConstants.EntityName.LAST_NAME,
+            CuKimPropertyConstants.EntityName.NAME_SUFFIX,
+            CuKimPropertyConstants.EntityAddress.LINE_1,
+            CuKimPropertyConstants.EntityAddress.LINE_2,
+            CuKimPropertyConstants.EntityAddress.LINE_3,
+            CuKimPropertyConstants.EntityAddress.CITY,
+            CuKimPropertyConstants.EntityAddress.STATE_PROVINCE_CODE,
+            CuKimPropertyConstants.EntityAddress.POSTAL_CODE,
+            CuKimPropertyConstants.SharedProperties.COUNTRY_CODE,
+            CuKimPropertyConstants.EntityPhone.PHONE_NUMBER,
+            CuKimPropertyConstants.EntityPhone.EXTENSION_NUMBER,
+            CuKimPropertyConstants.EntityEmail.EMAIL_ADDRESS
+    );
+
+    private UiDocumentService uiDocumentService;
+
+    @SuppressWarnings("rawtypes")
+    @Override
+    public BusinessObject getBusinessObject(Map fieldValues) {
+        // KFSPTS-19308 CU customization to allow inquiry by principalName
+        Person person = null;
+        if (fieldValues.containsKey("principalId")) {
+            person = getPersonService().getPerson(fieldValues.get("principalId").toString());
+        }
+
+        if (person == null && fieldValues.containsKey("principalName")) {
+            String principalName = fieldValues.get("principalName").toString();
+            if (StringUtils.isNotBlank(principalName)) {
+                person = getPersonService().getPersonByPrincipalName(principalName);
+            }
+        }
+
+        if (person != null && person instanceof PersonImpl) {
+            ((PersonImpl) person).populateMembers();
+        }
+        return person;
+
+    }
+
+    @Override
+    public List<Section> getSections(BusinessObject businessObject) {
```

Comment:
From your comment on the PR I understand that this code was present in the old base code but not in the new implementation. Should we report this to KualiCo?

Reply:
Just to clarify, this particular inquiry-section-modifying code is not was present in base code before. What was present before was the ability for the old document-style inquiry to unmask suppressed person fields to appropriate users. This custom inquiry-section-modifying code is trying to replicate that behavior using the regular inquiry setup.

We might have reported this issue to KualiCo already; there's a comment on KFSPTS-19335 stating that Request 12480 was created for it. Not sure if KualiCo created a FINP ticket for it yet, though. I do think it would be worth following up with KualiCo again on this, to see if they're changing the behavior of this field masking as part of their KIM Person rework.

Reply:
Thank you for clarifying this.

Reply:
https://kualico.atlassian.net/browse/FINP-6977 was closed as Won't Fix.